src/Symfony/Component/Mime/Address.php

```diff
@@ -33,20 +33,18 @@ final class Address
      */
     private const FROM_STRING_PATTERN = '~(?<displayName>[^<]*)<(?<addrSpec>.*)>[^>]*~';
 
-    private static EmailValidator $validator;
+    private static EmailValidator|EmailValidatorInterface $validator;
     private static IdnAddressEncoder $encoder;
 
     private string $address;
     private string $name;
 
-    public function __construct(string $address, string $name = '')
+    public function __construct(string $address, string $name = '', EmailValidatorInterface $customValidator = null)
```

Comment:
should be a https://github.com/egulias/EmailValidator/blob/4.x/src/Validation/EmailValidation.php IMHO